### ISSUE TITLE
AdsError: 6 (0X6, sorry, no error text [port 500])

### ISSUE THREAD

**Issue by bkchaluvadi** [Beckhoff-USA-Community/PackML_PLC_Example#6]
Hi , 

OS: Windows 11
twincat version: 4024.53
target system: UmRT_Default

I am trying to run this project on my local machine. The project build with no errors. But when I tried running the project, it shows an error.

`Init4\IO: Download Task Attrib: NC: new startup >>  AdsError: 6 (0X6, sorry, no error text [port 500]) << failed`

Info: I do not see any devices to disable in my project, only a single entry present in my Mapping.
<img width="1081" alt="PackMLProject" src="https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/0c408596-d163-4f8e-af7b-57daa0119301">
<img width="719" alt="PackMLProject2" src="https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/ad530708-4830-4ec7-9144-902906f24feb">
<img width="794" alt="PackMLProject3" src="https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/baf7d86e-c139-4df4-82e7-379a66cb3704">

**Comment by MarkLewis-Beckhoff** [comment]
Ads return code 6 is port not found and port 500 is the NC port for motion.

do you have the license setup for TF5000?. the project was built without hardware so you should not need to disable anything.

**Comment by bkchaluvadi** [comment]
I don't have the license for TF500. Aren't all licenses for the Beckhoff development environment free and renewed weekly? Additionally, after disabling "NC-Task 1 SAF", I encountered a different login error. While it successfully opened "Port_851", it faced subsequent issues.
<img width="799" alt="PackMLProject6" src="https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/11aa9dc3-6deb-4c62-8796-06a21c199089">

**Comment by MarkLewis-Beckhoff** [comment]
yes they are free to develop. just enable the motion and go to the license tab and click "7 days trial license"
![image](https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/53493185/9491ac59-2360-438b-86b5-9ad147869f50)

**Comment by bkchaluvadi** [comment]
Looks like it is already in activated state.
<img width="1038" alt="PackMLProject7" src="https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/fe402fc9-cde4-4c4a-981d-5946597e4bab">

**Comment by bkchaluvadi** [comment]
When I run the `Activate Configuration` with out disabling the `NC - Task 1 SAF`. I can see the following error on the `UmRT_Default `terminal

```
TwinCAT System: Source: TcOsSys;  Timestamp: 10/30/2023 4:30:05 PM 960 ms  Message: LoadExternalModules(): Error: LoadDrv for driver >TcNc< failed with error 0xc0000001 (STATUS_UNSUCCESSFUL)
Sending ams command >> Init4\IO: Download Task A
![PackMLProject8](https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/b90299f0-bd7f-4cb9-b333-22b60cc11404)
![PackMLProject8](https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/ef1b6f09-3a20-4d4c-9692-f44aea3c60a5)
ttrib: NC: new startup >> AdsError: 6 (0x6, sorry, no error text [port 500]) << failed!
TcSysSrvUm state: Config
```
![PackMLProject8](https://github.com/Beckhoff-USA-Community/PackML_PLC_Example/assets/144023762/5fd46ab4-6865-47fb-8a57-9ceaa598d470)

**Comment by MarkLewis-Beckhoff** [comment]
I am sorry this has gone on so long have you found what was the cause or is this still going on?

**Comment by LakeLiving** [comment]
I have the same problem and would be interested about the solution.

**Comment by RyanMontigny** [comment]
I just ran into this issue also...

we must install the following package for the user mode runtime to work with the NC: 
TwinCAT.XARUM.NCPTP

after installing that package, the port 500 issue was resolved on my system. I hope this helps!

**Comment by sgwong** [comment]
> I just ran into this issue also...
> 
> we must install the following package for the user mode runtime to work with the NC: TwinCAT.XARUM.NCPTP
> 
> after installing that package, the port 500 issue was resolved on my system. I hope this helps!

This works. thanks.
